### PR TITLE
feat: index contract init events and internal calls

### LINKED ISSUE #377: index `:contract_create_tx` events
Currently the events are logged only for contract calls transactions:
https://github.com/aeternity/ae_mdw/blob/ee3465822fecf346acc564e36f9e209faf8444f8/lib/ae_mdw/db/sync/transaction.ex#L82

### PR DESCRIPTION
**Pull request by jyeshe**
## Why
Resolve #377 

## Validation steps
elixir --sname aeternity@localhost -S mix test.integration test/integration/ae_mdw_web/controllers/contract_controller_test.exs

### CODE REVIEW

File: lib/ae_mdw/db/sync/transaction.ex [hunk 1]
```diff
@@ -82,11 +82,16 @@ defmodule AeMdw.Db.Sync.Transaction do
...
+      case type do
+        :contract_create_tx -> :aect_create_tx.contract_pubkey(tx)
+        :contract_call_tx -> :aect_call_tx.contract_pubkey(tx)
+        _other_tx -> nil
+      end
+
     contract_events_mutation =
-      if type == :contract_call_tx do
-        ct_pk = :aect_call_tx.contract_pubkey(tx)
+      if not is_nil(ct_pk) do
```
Comment (sborrazas) [inline lib/ae_mdw/db/sync/transaction.ex:93]: I think an `if ct_pk` should do here

File: lib/ae_mdw/contract.ex [hunk 2]
```diff
@@ -306,13 +306,18 @@ defmodule AeMdw.Contract do
     end
   end
 
-  def get_init_call_details(contract_pk, tx_rec, block_hash) do
+  def get_init_call_rec(contract_pk, tx_rec, block_hash) do
```
Comment (sborrazas) [inline lib/ae_mdw/contract.ex:309]: We should probably add typespecs to every new public function we create

**Review by sborrazas — CHANGES_REQUESTED** [review]

File: lib/ae_mdw/db/sync/transaction.ex (hunk 1 above)
Comment (jyeshe) [reply lib/ae_mdw/db/sync/transaction.ex]: glad you prefer shorter too

File: lib/ae_mdw/contract.ex (hunk 2 above)
Comment (jyeshe) [reply lib/ae_mdw/contract.ex:265]: added for this and remaining functions of the module to enable credo/spec check

**Review by thepiwo — APPROVED** [review]

**Review by sborrazas — APPROVED** [review]

**Pull request merged**